csrc/fusion_segmenter.h

```diff
@@ -497,7 +497,8 @@ struct SegmentCandidateFinderOptions {
   bool run_combine_reductions = true;
   bool run_herrmann_merge = true;
   bool run_final_merge = true;
-  bool only_segment_resharding_exprs = false;
+  bool (*custom_should_merge_groups)(SegmentedGroup*, SegmentedGroup*) =
```

Comment:
You may want to consider std::function, which may be slower but more flexible for the future (e.g. supporting lambda).

Reply:
right, thanks